README
Tk-ForDummies-Graph

This is Tk::ForDummies::Graph, a package to generate charts lines, pie and bars.
See the documentation for histories and more informations.


INSTALLATION

To install this module, run the following commands:

	perl Makefile.PL
	make
	make test
	make install

Alternatively, to install with Module::Build, you can use the following commands:

	perl Build.PL
	./Build
	./Build test
	./Build install

DEPENDANCIES

    Tk::ForDummies::Graph needs Tk installed
   
- On Debian, you can install Tk via CPAN or apt-get

  apt-get install perl-tk


SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the
perldoc command.

    perldoc Tk::ForDummies::Graph
    perldoc Tk::ForDummies::Graph::Lines
    perldoc Tk::ForDummies::Graph::Splines
    perldoc Tk::ForDummies::Graph::Bars
    perldoc Tk::ForDummies::Graph::Areas
    perldoc Tk::ForDummies::Graph::Pie
    perldoc Tk::ForDummies::Graph::Points
    perldoc Tk::ForDummies::Graph::Boxplots
    perldoc Tk::ForDummies::Graph::FAQ


You can also look for information at:

    RT, CPAN's request tracker
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Tk-ForDummies-Graph

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/Tk-ForDummies-Graph

    CPAN Ratings
        http://cpanratings.perl.org/d/Tk-ForDummies-Graph

    Search CPAN
        http://search.cpan.org/dist/Tk-ForDummies-Graph/


COPYRIGHT AND LICENCE

Copyright (C) 2009 Djibril Ousmanou

This program is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.



Changes
Revision history for Tk-ForDummies-Graph

1.13 19/06/2010 22:22:23
  Tk-ForDummies-Graph is now an extension of Tk::Canvas::GradientColor which allow you to 
  to set a gradient background color in your graph. Then, you can use all methods of this module.
  Bugs has been fixed : 
    -Resizing the widget does not remove the items that have been added on the graph.
    -The bug binding keys has been fixed.
  -highlightthickness sets to 0 by default to remove border focus rectangle.

1.12 22/05/2010 19:19:14
  Bugs has been fixed in Tk::ForDummies::Graph::Mixed.
  Bugs has been fixed in points marker legend.
  -outlinearea and -outlinebar options has been added in Tk::ForDummies::Graph::Mixed, Tk::ForDummies::Graph::Areas and Tk::ForDummies::Graph::Bars.
  Documentation updated.

1.11 21/05/2010 23:01:43
  really removed -smoothline option from Tk::ForDummies::Graph::Lines.
  Tk::ForDummies::Graph::Mixed added.
  New display_values method added for Tk::ForDummies::Graph::Lines, Points, Splines, Areas and Mixed.
  New enabled_automatic_redraw and disabled_automatic_redraw methods added for all Tk::ForDummies::Graph modules.
  -dash option added to Tk::ForDummies::Graph::Lines. 
  Modules code updated.
  POD documentation updated.
  Tk::ForDummies::Graph::FAQ updated : question about Scrolled and Tk::ForDummies::Graph added.
  Screenshots added in examples.
   
1.10 18/07/2009 00:22:43
  Tk::ForDummies::Graph::Boxplots 
    The same algorithm as Excel and type 7 R package is used for the quantile Q1 et Q3 calculation. 
    new boxplot_information method added.
   
  Warning fixed in all add_data method (Tk::ForDummies::Graph::*) when legend is not defined.
  CPAN documentation updated.

1.09 16/07/2009 20:26:56
  The calcul of quantile (Q1, Q3) was wrong for impair numbers of data values
  in Tk::ForDummies::Graph::Boxplots. The bug is now fixed and the calcul 
  is the same as in Excel, and in R language it is the equivalent of quantile type 7.


1.08 07/07/2009 21:45:08
  New options in Tk::ForDummies::Graph::Lines module to create lines chart as Bezier curve :
    -bezier replaces -smoothline (deprecated) and allows to create lines chart as Bezier curve.
  New option -spline, the curve crosses by all points.

1.07 13/05/2009
  New options -titleposition, -xvalueview and -yvalueview.
  
1.06 13/04/2009 16:54:30
  fix bugs in Tk-ForDummies-Graph modules (legend, ...).
  Tk::ForDummies::Graph::Boxplots updated.
  
1.05    11/04/2009 01:00:22 
  Tk::ForDummies::Graph::Splines added.
  Tk::ForDummies::Graph::FAQ added.
  Tk::ForDummies::Graph::Boxplots added.
  Tk::ForDummies::Graph::Points added.
  
  New option -smoothline in Tk::ForDummies::Graph::Lines
  
  fix bug :
    - If you don't have a value for a point in a dataset, you can use undef, and the point will be really skipped.
    
1.04    01/04/2009 Fix bug installation, pod documentations and script bugs.

1.03    31/03/2009 Tk::ForDummies::Graph::Areas added.

1.02    30/03/2009 Tk::ForDummies::Graph::Pie updated.
  option -startangle have been added
  
1.01    30/03/2009 Change Read me.

1.00    24/03/2009 First version for CPAN.



demo/Mixed.pl
#!/usr/bin/perl
use strict;
use warnings;
use Tk;
use Tk::ForDummies::Graph::Mixed;
use Tk::Pane;

my $mw = new MainWindow(
  -title      => 'Tk::ForDummies::Graph::Mixed',
  -background => 'white',
);

my @types = ( 'areas', 'bars', 'lines', 'points', 'bars', 'dashlines' );
my $GraphDummies = $mw->Mixed(
  -title      => 'Tk::ForDummies::Graph::Mixed',
  -xlabel     => 'X Label',
  -ylabel     => 'Y Label',
  -background => '#D0D0FF',
  -linewidth  => 2,
  -typemixed  => \@types,
  -markers    => [ 3, 5, 6 ],

  -longticks => 1,
)->pack(qw / -fill both -expand 1 /);

my @data = (
  [ '1st', '2nd', '3rd', '4th', '5th', '6th', '7th', '8th', '9th' ],
  [ 90,    29,    25,    6,     -20,   1,     1,     3,     4 ],
  [ 15,    10,    5,     2,     3,     5,     7,     9,     12 ],
  [ 1,     2,     12,    6,     3,     5,     1,     23,    5 ],
  [ 15,    12,    24,    33,    19,    8,     6,     15,    21 ],
  [ 15,    2,     52,    6,     3,     17.5,  1,     43,    10 ],
  [ 30,    2,     5,     6,     3,     1.5,   1,     3,     4 ],
  [ 24,    12,    35,    20,    13,    31.5,  41,    6,     25 ],

);

# Add a legend to the graph
my @Legends = @types;
$GraphDummies->set_legend(
  -title       => 'Title legend',
  -data        => [ 'legend 1', 'legend 2', 'legend 3', 'legend 4', 'legend 5', 'legend 6', 'legend 7', ],
  -titlecolors => 'blue',
);

# Add help identification
$GraphDummies->set_balloon();

# Create the graph
$GraphDummies->plot( \@data );

# background order wanted
$GraphDummies->display_order( [qw/ areas bars lines  linespoints points /] );

MainLoop();
